sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -114,8 +137,39 @@ SamplingResult shouldSample(
       SpanKind spanKind,
       Attributes attributes,
       List<LinkData> parentLinks) {
-    return fixedRateSampler.shouldSample(
-        parentContext, traceId, name, spanKind, attributes, parentLinks);
+    // Incrementing requests first ensures sample / borrow rate are positive.
+    statistics.requests.increment();
+    SamplingResult result =
+        reservoirSampler.shouldSample(
+            parentContext, traceId, name, spanKind, attributes, parentLinks);
+    if (result.getDecision() != SamplingDecision.DROP) {
+      // We use the result from the reservoir sampler if it worked.
+      if (borrowing) {
+        statistics.borrowed.increment();
+      }
+      statistics.sampled.increment();
+      return result;
+    }
+    result =
```

Comment:
I'm confused (maybe it's the ridiculous heat in Portland right now). do I read this correctly that if the reservoir sampler says "DROP", you get a 2nd opinion from the fixedRateSampler?